src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -47,54 +52,308 @@ export class PermissionsTableComponent implements OnInit {
+              });
+              break;
+          }
+        });
+        break;
+      case 'project':
+        this.route.paramMap.subscribe((params: ParamMap) => {
+          let orgCode: string = params.get('organizationCode');
+          let proCode: string = params.get('projectCode');
+          switch (this.entityType) {
+            case 'users':
+              this.entities[index]['roles'].forEach(role => {
+                this.removeProjectUsersGroupsRole(orgCode, proCode, role, [this.entities[index]['name']], []);
+              });
+              break;
+            case 'groups':
+              this.entities[index]['roles'].forEach(role => {
+                this.removeProjectUsersGroupsRole(orgCode, proCode, role, [], [this.entities[index]['name']]);
+              });
+              break;
+          }
+        });
+    }
+    this.otherEntities.push(this.entities[index]['name']);
+    this.entities.splice(index, 1);
+  }
+
+  public changePermission(entityName: string, role: string, event) {
+    let addPermision: boolean = event.target.checked;
+    this.route.paramMap.subscribe((params: ParamMap) => {
+      let orgCode: string = params.get('organizationCode');
+      switch (this.resourceType) {
+        case 'organization':
+          switch (this.entityType) {
+            case 'users':
+              if (addPermision) {
+                  this.addOrganizationUsersGroupsRole(orgCode, role, [entityName], []);
+              } else {
+                  this.removeOrganizationUsersGroupsRole(orgCode, role, [entityName], []);
+              }
+              break;
+            case 'groups':
+              if (addPermision) {
+                  this.addOrganizationUsersGroupsRole(orgCode, role, [], [entityName]);
+              } else {
+                  this.removeOrganizationUsersGroupsRole(orgCode, role, [], [entityName]);
+              }
+          }
+          break;
+        case 'project':
+          let proCode: string = params.get('projectCode');
+          switch (this.entityType) {
+            case 'users':
+              if (addPermision) {
+                this.route.paramMap.subscribe((params: ParamMap) => {
+                  this.addProjectUsersGroupsRole(orgCode, proCode, role, [entityName], []);
+                });
+              } else {
+                this.route.paramMap.subscribe((params: ParamMap) => {
+                  this.removeProjectUsersGroupsRole(orgCode, proCode, role, [entityName], []);
+                });
+              }
+              break;
+            case 'groups':
+              if (addPermision) {
+                  this.addProjectUsersGroupsRole(orgCode, proCode, role, [], [entityName]);
+              } else {
+                  this.removeProjectUsersGroupsRole(orgCode, proCode, role, [], [entityName]);
+              }
+          }
+          break;
+      }
+    });
+  }
+
+  private getEntitiesForUserOrGroup(userOrGroup: string, roles: Role[]): Entity[] {
+    let arr = [];
```

Comment:
What does this variable represent? Try to use more self-explaining names even though they are longer.